### PR TITLE
[db] Soft delete entries in the `db_volume_snapshot` table

### PR DESCRIPTION
**Pull request by andrew-farries**
## Description

Replace the existing hard-deletion of rows in the `db_volume_snapshot` table with soft-deletion.

This will allow `db-sync` to propagate deletions between clusters.

## Related Issue(s)

Follow up to https://github.com/gitpod-io/gitpod/pull/13348

Part of #9198 

## How to test

Unit tests

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: components/gitpod-db/src/typeorm/workspace-db-impl.ts [hunk 1]
```diff
@@ -764,7 +764,11 @@ export abstract class AbstractTypeORMWorkspaceDBImpl implements WorkspaceDB {
 
     public async findVolumeSnapshotById(volumeSnapshotId: string): Promise<VolumeSnapshot | undefined> {
         const volumeSnapshots = await this.getVolumeSnapshotRepo();
-        return volumeSnapshots.findOne(volumeSnapshotId);
+        return volumeSnapshots
+            .createQueryBuilder("vs")
+            .where("vs.deleted = 0")
+            .andWhereInIds(volumeSnapshotId)
```
Comment (easyCZ) [inline components/gitpod-db/src/typeorm/workspace-db-impl.ts:770]: What's the difference between `findOne` and `andWhereInIds`?

**Review by easyCZ — APPROVED** [review]
LGTM, tho not an expert on our TypeORM models

/hold for Q

File: components/gitpod-db/src/typeorm/workspace-db-impl.ts (hunk 1 above)
Comment (geropl) [reply components/gitpod-db/src/typeorm/workspace-db-impl.ts:770]: Wondering the same. Using `.andWhere("vs.id = :id", { id: volumneSnapshotId })` would be more direct, and require less TypeORM knowledge.

Comment (andrew-farries) [reply components/gitpod-db/src/typeorm/workspace-db-impl.ts]: Thanks, I've rebased to make this change.

**Pull request merged**